### PR TITLE
Enhance CNVM index

### PR DESCRIPTION
**Pull request by jeniawhite**
### Summary of your changes
Enhancing the CNVM index with additional data from the instance that is being scanned.
Added cloud, network, host, and security group information to the index.
Note that the cloud and host data is regarding scanned instances and not the scanner instance.
This is why we did not use cloud and host processors because they collect information for the scanner machine.

Example of event:
```json
{
  "_index": ".ds-logs-cloud_security_posture.vulnerabilities-default-2023.07.16-000001",
  "_id": "SLZNX4kBz2hMpetnpF4Z",
  "_version": 1,
  "_score": 0,
  "_source": {
    "agent": {
      "name": "ip-172-31-32-84.eu-west-1.compute.internal",
      "id": "1f859814-361a-4693-8717-fdbd2b9c45c7",
      "ephemeral_id": "39524697-5b09-4a5e-bfdc-abdd1bc0983f",
      "type": "cloudbeat",
      "version": "8.9.0"
    },
    "package": {
      "path": "var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/81/fs/usr/local/bin/trivy",
      "fixed_version": "2.8.2-beta.1",
      "name": "github.com/docker/distribution",
      "type": "gobinary",
      "version": "v2.8.1+incompatible"
    },
    "resource": {
      "name": "cloudbeat-tf-5jA-1",
      "id": "0d931130a4e7a3d2b"
    },
    "elastic_agent": {
      "id": "1f859814-361a-4693-8717-fdbd2b9c45c7",
      "version": "8.9.0",
      "snapshot": false
    },
    "vulnerability": {
      "severity": "HIGH",
      "package": {
        "fixed_version": "2.8.2-beta.1",
        "name": "github.com/docker/distribution",
        "version": "v2.8.1+incompatible"
      },
      "description": "A flaw was found in the `/v2/_catalog` endpoint in distribution/distribution, which accepts a parameter to control the maximum number of records returned (query string: `n`). This vulnerability allows a malicious user to submit an unreasonably large value for `n,` causing the allocation of a massive string array, possibly causing a denial of service through excessive use of memory.",
      "title": "DoS from malicious API request",
      "classification": "CVSS",
      "data_source": {
        "ID": "ghsa",
        "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Ago",
        "Name": "GitHub Security Advisory Go"
      },
      "cwe": [
        "CWE-770"
      ],
      "reference": "https://avd.aquasec.com/nvd/cve-2023-2253",
      "score": {
        "version": "3.1",
        "base": 7.5
      },
      "report_id": 1689519799,
      "scanner": {
        "vendor": "Trivy",
        "version": "v0.35.0"
      },
      "id": "CVE-2023-2253",
      "enumeration": "CVE",
      "category": "lang-pkgs",
      "cvss": {
        "redhat": {
          "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
          "V3Score": 6.5
        },
        "nvd": {
          "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
          "V3Score": 6.5
        },
        "ghsa": {
          "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
          "V3Score": 7.5
        }
      },
      "class": "lang-pkgs",
      "published_date": "2023-06-06T20:15:00Z"
    },
    "network": {
      "private_ip": "10.0.3.115",
      "mac_addresses": [
        "0a:38:d6:ba:7b:57",
        "0a:3d:a1:79:86:35"
      ]
    },
    "cloud": {
      "availability_zone": "eu-west-1c",
      "instance": {
        "name": "cloudbeat-tf-5jA-1",
        "id": "0d931130a4e7a3d2b"
      },
      "provider": "aws",
      "service": {
        "name": "AWS EC2"
      },
      "machine": {
        "launch_time": "2023-01-11T10:16:16Z",
        "image": "ami-0ae92cf6a0a5d24d0",
        "type": "t3.small",
        "authentication": {}
      },
      "region": "eu-west-1",
      "account": {
        "name": "elastic-security-cloud-security-dev",
        "id": "704479110758"
      },
      "tags": {
        "org": "security",
        "aws:eks:cluster-name": "cloudbeat-tf-5jA",
        "kubernetes.io/cluster/cloudbeat-tf-5jA": "owned",
        "eks:cluster-name": "cloudbeat-tf-5jA",
        "project": "daily-env",
        "team": "cloud-security-posture",
        "aws:ec2launchtemplate:version": "1",
        "k8s.io/cluster-autoscaler/enabled": "true",
        "k8s.io/cluster-autoscaler/cloudbeat-tf-5jA": "owned",
        "Name": "cloudbeat-tf-5jA-1",
        "division": "engineering",
        "aws:autoscaling:groupName": "eks-cloudbeat-tf-5jA-1-20230111101512073200000014-c6c2d029-ce4f-c677-2264-ea672f59fa8f",
        "aws:ec2:fleet-id": "fleet-161f3d96-aaac-4e2d-86ba-a32868609483",
        "eks:nodegroup-name": "cloudbeat-tf-5jA-1-20230111101512073200000014",
        "aws:ec2launchtemplate:id": "lt-08bdd2639abf1e9f2"
      }
    },
    "security": {
      "security_groups": [
        {
          "group_id": "sg-0066ee97b29f7a65b",
          "group_name": "cloudbeat-tf-5jA-node-20230111100434286600000006"
        },
        {
          "group_id": "sg-0efec4e7bfeb573d1",
          "group_name": "eks-cluster-sg-cloudbeat-tf-5jA-821802285"
        },
        {
          "group_id": "sg-081b6075cf7a6671e",
          "group_name": "cloudbeat-tf-5jA_120230111100433256600000005"
        }
      ]
    },
    "@timestamp": "2023-07-16T15:24:21.883Z",
    "cloudbeat": {
      "commit_sha": "2a9a2de4d2cac154d4bb7fddf453abd7add47c17",
      "commit_time": "2023-07-16T13:35:59Z",
      "version": "8.9.0"
    },
    "ecs": {
      "version": "8.6.0"
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "cloud_security_posture.vulnerabilities"
    },
    "host": {
      "os": {
        "platform": "Linux/UNIX"
      },
      "name": "cloudbeat-tf-5jA-1",
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "sequence": 1689519799,
      "ingested": "2023-07-16T15:24:31Z",
      "created": "2023-07-16T15:24:21.883016733Z",
      "kind": "state",
      "id": "bf19c365-8686-4526-90e6-14386f3fc3fa",
      "category": [
        "vulnerability"
      ],
      "type": [
        "info"
      ],
      "dataset": "cloud_security_posture.vulnerabilities",
      "outcome": "success"
    }
  },
  "fields": {
    "vulnerability.package.version": [
      "v2.8.1+incompatible"
    ],
    "event.category": [
      "vulnerability"
    ],
    "elastic_agent.version": [
      "8.9.0"
    ],
    "vulnerability.cvss.nvd.V3Vector": [
      "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H"
    ],
    "vulnerability.cvss.ghsa.V3Vector": [
      "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
    ],
    "cloud.tags.aws:eks:cluster-name": [
      "cloudbeat-tf-5jA"
    ],
    "vulnerability.scanner.vendor": [
      "Trivy"
    ],
    "cloud.tags.eks:nodegroup-name": [
      "cloudbeat-tf-5jA-1-20230111101512073200000014"
    ],
    "vulnerability.data_source.ID": [
      "ghsa"
    ],
    "cloud.availability_zone": [
      "eu-west-1c"
    ],
    "package.type": [
      "gobinary"
    ],
    "cloud.tags.aws:autoscaling:groupName": [
      "eks-cloudbeat-tf-5jA-1-20230111101512073200000014-c6c2d029-ce4f-c677-2264-ea672f59fa8f"
    ],
    "cloud.tags.k8s.io/cluster-autoscaler/enabled": [
      "true"
    ],
    "vulnerability.data_source.URL": [
      "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Ago"
    ],
    "agent.name": [
      "ip-172-31-32-84.eu-west-1.compute.internal"
    ],
    "host.name": [
      "cloudbeat-tf-5jA-1"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "cloud.tags.aws:ec2launchtemplate:version": [
      "1"
    ],
    "event.kind": [
      "state"
    ],
    "cloud.tags.project": [
      "daily-env"
    ],
    "event.outcome": [
      "success"
    ],
    "package.version": [
      "v2.8.1+incompatible"
    ],
    "cloud.region": [
      "eu-west-1"
    ],
    "package.path": [
      "var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/81/fs/usr/local/bin/trivy"
    ],
    "vulnerability.title": [
      "DoS from malicious API request"
    ],
    "cloud.machine.image": [
      "ami-0ae92cf6a0a5d24d0"
    ],
    "data_stream.type": [
      "logs"
    ],
    "cloud.tags.org": [
      "security"
    ],
    "cloud.tags.k8s.io/cluster-autoscaler/cloudbeat-tf-5jA": [
      "owned"
    ],
    "host.architecture": [
      "x86_64"
    ],
    "cloud.provider": [
      "aws"
    ],
    "cloud.machine.type": [
      "t3.small"
    ],
    "cloudbeat.version": [
      "8.9.0"
    ],
    "cloud.tags.Name": [
      "cloudbeat-tf-5jA-1"
    ],
    "agent.id": [
      "1f859814-361a-4693-8717-fdbd2b9c45c7"
    ],
    "cloud.service.name": [
      "AWS EC2"
    ],
    "ecs.version": [
      "8.6.0"
    ],
    "event.created": [
      "2023-07-16T15:24:21.883Z"
    ],
    "cloud.machine.launch_time": [
      "2023-01-11T10:16:16Z"
    ],
    "cloud.tags.aws:ec2:fleet-id": [
      "fleet-161f3d96-aaac-4e2d-86ba-a32868609483"
    ],
    "agent.version": [
      "8.9.0"
    ],
    "vulnerability.reference": [
      "https://avd.aquasec.com/nvd/cve-2023-2253"
    ],
    "vulnerability.cvss.nvd.V3Score": [
      6.5
    ],
    "cloudbeat.commit_sha": [
      "2a9a2de4d2cac154d4bb7fddf453abd7add47c17"
    ],
    "vulnerability.package.fixed_version": [
      "2.8.2-beta.1"
    ],
    "vulnerability.enumeration": [
      "CVE"
    ],
    "vulnerability.cvss.redhat.V3Vector": [
      "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H"
    ],
    "vulnerability.published_date": [
      "2023-06-06T20:15:00Z"
    ],
    "resource.name": [
      "cloudbeat-tf-5jA-1"
    ],
    "vulnerability.package.name": [
      "github.com/docker/distribution"
    ],
    "cloud.tags.aws:ec2launchtemplate:id": [
      "lt-08bdd2639abf1e9f2"
    ],
    "event.sequence": [
      1689519799
    ],
    "cloud.instance.id": [
      "0d931130a4e7a3d2b"
    ],
    "agent.type": [
      "cloudbeat"
    ],
    "vulnerability.id": [
      "CVE-2023-2253"
    ],
    "vulnerability.score.version": [
      "3.1"
    ],
    "cloud.tags.kubernetes.io/cluster/cloudbeat-tf-5jA": [
      "owned"
    ],
    "package.name": [
      "github.com/docker/distribution"
    ],
    "vulnerability.score.base": [
      7.5
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "network.mac_addresses": [
      "0a:38:d6:ba:7b:57",
      "0a:3d:a1:79:86:35"
    ],
    "vulnerability.report_id": [
      "1.689519799E9"
    ],
    "cloud.tags.division": [
      "engineering"
    ],
    "cloud.account.name": [
      "elastic-security-cloud-security-dev"
    ],
    "vulnerability.classification": [
      "CVSS"
    ],
    "elastic_agent.id": [
      "1f859814-361a-4693-8717-fdbd2b9c45c7"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "vulnerability.class": [
      "lang-pkgs"
    ],
    "security.security_groups.group_id": [
      "sg-0066ee97b29f7a65b",
      "sg-0efec4e7bfeb573d1",
      "sg-081b6075cf7a6671e"
    ],
    "vulnerability.cwe": [
      "CWE-770"
    ],
    "vulnerability.scanner.version": [
      "v0.35.0"
    ],
    "resource.id": [
      "0d931130a4e7a3d2b"
    ],
    "vulnerability.category": [
      "lang-pkgs"
    ],
    "network.private_ip": [
      "10.0.3.115"
    ],
    "event.ingested": [
      "2023-07-16T15:24:31.000Z"
    ],
    "@timestamp": [
      "2023-07-16T15:24:21.883Z"
    ],
    "vulnerability.severity": [
      "HIGH"
    ],
    "security.security_groups.group_name": [
      "cloudbeat-tf-5jA-node-20230111100434286600000006",
      "eks-cluster-sg-cloudbeat-tf-5jA-821802285",
      "cloudbeat-tf-5jA_120230111100433256600000005"
    ],
    "cloud.account.id": [
      "704479110758"
    ],
    "host.os.platform": [
      "Linux/UNIX"
    ],
    "event.type": [
      "info"
    ],
    "data_stream.dataset": [
      "cloud_security_posture.vulnerabilities"
    ],
    "package.fixed_version": [
      "2.8.2-beta.1"
    ],
    "vulnerability.data_source.Name": [
      "GitHub Security Advisory Go"
    ],
    "agent.ephemeral_id": [
      "39524697-5b09-4a5e-bfdc-abdd1bc0983f"
    ],
    "vulnerability.cvss.redhat.V3Score": [
      6.5
    ],
    "vulnerability.cvss.ghsa.V3Score": [
      7.5
    ],
    "vulnerability.description": [
      "A flaw was found in the `/v2/_catalog` endpoint in distribution/distribution, which accepts a parameter to control the maximum number of records returned (query string: `n`). This vulnerability allows a malicious user to submit an unreasonably large value for `n,` causing the allocation of a massive string array, possibly causing a denial of service through excessive use of memory."
    ],
    "vulnerability.description.text": [
      "A flaw was found in the `/v2/_catalog` endpoint in distribution/distribution, which accepts a parameter to control the maximum number of records returned (query string: `n`). This vulnerability allows a malicious user to submit an unreasonably large value for `n,` causing the allocation of a massive string array, possibly causing a denial of service through excessive use of memory."
    ],
    "event.id": [
      "bf19c365-8686-4526-90e6-14386f3fc3fa"
    ],
    "cloudbeat.commit_time": [
      "2023-07-16T13:35:59.000Z"
    ],
    "cloud.tags.team": [
      "cloud-security-posture"
    ],
    "event.dataset": [
      "cloud_security_posture.vulnerabilities"
    ],
    "cloud.instance.name": [
      "cloudbeat-tf-5jA-1"
    ],
    "cloud.tags.eks:cluster-name": [
      "cloudbeat-tf-5jA"
    ]
  }
}
```

### Related Issues
- Fixes: https://github.com/elastic/cloudbeat/issues/983

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: vulnerability/events_creator.go [hunk 1]
```diff
@@ -57,6 +58,59 @@ type Vulnerability struct {
...
+	Platform *string `json:"platform,omitempty"`
+}
+
+type Service struct {
+	Name string `json:"name,omitempty"`
+}
+
+type Machine struct {
+	Type       string     `json:"type,omitempty"`
+	Key        AuthInfo   `json:"key,omitempty"`
```
Comment (amirbenun) [inline vulnerability/events_creator.go:105]: This will create `cloud.machine.key.key`, please rename to `authentication`

**Review by amirbenun — COMMENTED** [review]
Looks good, can you please print a single vulnerability?

File: vulnerability/events_creator.go [hunk 2]
```diff
@@ -220,6 +310,33 @@ func getIdentifierType(id string) string {
...
+	securityGroups := make([]SecurityGroupInfo, len(groups))
+	for i, group := range groups {
+		securityGroups[i] = SecurityGroupInfo{
+			GroupId:   group.GroupId,
+			GroupName: group.GroupName,
+		}
+	}
+	return securityGroups
+}
+
```
Comment (uri-weisman) [inline vulnerability/events_creator.go:339]: just wondering if maybe those methods should reside under the ec2 provider.

**Comment by jeniawhite** [comment]
@amirbenun 
Please review fields added/removed due to this [document](https://docs.google.com/document/d/1UFWt0raEohAN6qJRGyGDZqBdLj7whRRddFqvo0lYK8I/edit).

**Review by amirbenun — APPROVED** [review]
@opauloh, we want to deprecate a few fields `vulnerability.class`, `resource.id`, `resource.name`.
Can you please check our FE is using any of them?

**Comment by opauloh** [comment]
> @opauloh, we want to deprecate a few fields `vulnerability.class`, `resource.id`, `resource.name`. Can you please check our FE is using any of them?

@amirbenun the `vulnerability.class` field is currently not being used. 

The fields `resource.id`, `resource.name` are widely used across all the CNVM pages, so we need to coordinate the changes on the Kibana side and also make sure to consider backward compatibility. 

image

image

image

@nick-alayil, in that case, are we going to review the naming in Kibana according to the new fields?

cc @kfirpeled for awareness and prioritization

**Comment by amirbenun** [comment]
Thanks @opauloh, so as a first step we will just add the new fields and use them in kibana instead.
Since we are in beta, I think we can remove completely the deprecated fields.
This will affect customer in the following way.
- Users who upgraded their stack version but not the agent will suffer the most as the UI will show blanks resource.id and resource.name.
- Users who upgraded their stack and agents will get that behavior for old findings only.

@nick-alayil what do you think? Do real users exploring CNVM? If not I would make this change before CNVM goes GA.

**Comment by oren-zohar** [comment]
> * Users who upgraded their stack version but not the agent will suffer the most as the UI will show blanks resource.id and resource.name.

In this case, we can create a migration flow like we already had in other cases, that will map these fields to the right place. I'm not sure if it was an ingest pipeline or something else in Kibana itself but I do remember we had something like that already @orouz @kfirpeled

**Pull request merged**